x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -232,19 +260,19 @@ synchronized AsyncStatusResponse toStatusResponse(String asyncExecutionId, long
             // include clusters in the status if present and not Clusters.EMPTY (the case for local searches only)
             clustersInStatus = clusters;
         }
-        if (finalResponse != null) {
+        if (latestFullResponse != null) {
             return new AsyncStatusResponse(
                 asyncExecutionId,
-                false,
-                false,
+                frozen == false,
+                frozen == false,
```

Comment:
I still get confused about this: should we look at isPartial for the isPartial flag?

Reply:
Yes, you are correct. Good catch. Fixed in next push.